src/memphis.ts

```diff
@@ -11,765 +11,573 @@
-    timeoutMs = 15000,
-  }: {
-    host: string;
-    managementPort?: number;
-    tcpPort?: number;
-    dataPort?: number;
-    username: string;
-    connectionToken: string;
-    reconnect?: boolean;
-    maxReconnect?: number;
-    reconnectIntervalMs?: number;
-    timeoutMs?: number;
-  }): Promise<void> {
-    return new Promise((resolve, reject) => {
-      this.host = this._normalizeHost(host);
-      this.managementPort = managementPort;
-      this.tcpPort = tcpPort;
-      this.dataPort = dataPort;
-      this.username = username;
-      this.connectionToken = connectionToken;
-      this.reconnect = reconnect;
-      this.maxReconnect = maxReconnect > 9 ? 9 : maxReconnect;
-      this.reconnectIntervalMs = reconnectIntervalMs;
-      this.timeoutMs = timeoutMs;
-
-      this.client.connect(this.tcpPort, this.host, () => {
-        this.client.write(
-          JSON.stringify({
-            username: username,
-            broker_creds: connectionToken,
-            connection_id: this.connectionId,
-          })
-        );
-
-        this.client.on("data", async (data) => {
-          let message: IMessage;
-          try {
-            message = JSON.parse(data.toString());
-          } catch (ex) {
-            return reject(data.toString());
-          }
-          this.connectionId = message.connection_id;
-          this.isConnectionActive = true;
-          this.reconnectAttempts = 0;
-
-          if (message.access_token) {
-            this.accessToken = message.access_token;
-            this._keepAcessTokenFresh(message.access_token_exp);
-          }
-
-          if (message.ping_interval_ms)
-            this._pingServer(message.ping_interval_ms);
-
-          if (!this.natsConnection) {
+    private isConnectionActive: boolean;
+    public connectionId: string;
+    public host: string;
+    public port: number;
+    public username: string;
+    private connectionToken: string;
+    private reconnect: boolean;
+    private maxReconnect: number;
+    private reconnectIntervalMs: number;
+    private timeoutMs: number;
+    public brokerConnection: any;
+    public brokerManager: any;
+    public brokerStats: any;
+    public retentionTypes!: IRetentionTypes;
+    public storageTypes!: IStorageTypes;
+    public JSONC: any;
+
+    constructor() {
+        this.isConnectionActive = false;
+        this.host = '';
+        this.port = 6666;
+        this.username = '';
+        this.connectionToken = '';
+        this.reconnect = true;
+        this.maxReconnect = 3;
+        this.reconnectIntervalMs = 200;
+        this.timeoutMs = 15000;
+        this.brokerConnection = null;
+        this.brokerManager = null;
+        this.brokerStats = null;
+        this.retentionTypes = retentionTypes;
+        this.storageTypes = storageTypes;
+        this.JSONC = broker.JSONCodec();
+        const genRanHex = (size) => [...Array(size)].map(() => Math.floor(Math.random() * 16).toString(16)).join('');
```

Comment:
call this function generate connectionID and make it a private function inside this class